### PR TITLE
lean 3.2.0 (new formula)

### PR DESCRIPTION
**Pull request by tchajed**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/lean.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class Lean < Formula
+  desc "Theorem prover"
+  homepage "https://leanprover.github.io/"
+  url "https://github.com/leanprover/lean/archive/v3.2.0.tar.gz"
+  sha256 "2668e075a9bdf4270d75fd60181078b33c078c7288b5bb8ad795819d8d5272cf"
+  head "https://github.com/leanprover/lean.git"
+
+  depends_on "cmake" => :build
+  depends_on "gmp"
+  depends_on "jemalloc" => :recommended
```
Comment (fxcoudert) [inline Formula/lean.rb]: We try to have formulas with the fewer options possible, so many people can benefit from our prebuilt binaries (bottles). Let's make `jemalloc` mandatory.

File: Formula/lean.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  depends_on "gperftools" => :optional
```
Comment (fxcoudert) [inline Formula/lean.rb]: What does gperftools bring in terms of functionality? Maybe we want it to be default, too? It's a relatively small dependency.

File: Formula/lean.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  def install
+    mkdir_p "build/release"
```
Comment (fxcoudert) [inline Formula/lean.rb]: Does it really need to be in a sub-sub-folder? Can't we do:

```
mkdir "build" do
  system "cmake", "../src", *std_cmake_args
  system "make", "install"
end
```

File: Formula/lean.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      inductive tree (α : Type) : Type
+      | node : α → list tree → tree
+
+      example (a b : Prop) : a ∧ b -> b ∧ a :=
+      begin
+          intro h, cases h,
+          split, repeat { assumption }
+      end
+    EOS
+    system "#{bin}/lean", testpath/"hello.lean"
```
Comment (fxcoudert) [inline Formula/lean.rb]: Test looks nice, but please check its output.

**Review by fxcoudert — CHANGES_REQUESTED** [review]
Thanks for the pull request!

File: Formula/lean.rb (hunk 2 above)
Comment (tchajed) [reply Formula/lean.rb]: lean uses jemalloc first, then tcmalloc from gperftools, then normal malloc; requiring jemalloc obviates the need for this at all

File: Formula/lean.rb (hunk 3 above)
Comment (tchajed) [reply Formula/lean.rb]: Thanks for the simplification!

File: Formula/lean.rb (hunk 4 above)
Comment (tchajed) [reply Formula/lean.rb]: There actually is no output; lean just verifies the input, which means it type checks the file and exits with a status of 0.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
@brewtestbot test this please

**Comment by fxcoudert** [comment]
Updated the `cmake` invocation to a slightly more usual setting.

**Comment by fxcoudert** [comment]
Many thanks @tchajed for the pull request!

**Pull request closed without merging**